src/Orleans/Core/Exceptions.cs

```diff
@@ -207,5 +207,29 @@ protected ClientNotAvailableException(SerializationInfo info, StreamingContext c
         { }
 #endif
     }
+
+    /// <summary>
+    /// Indicates that a <see cref="GrainReference"/> was not bound to the runtime before being used.
+    /// </summary>
+    [Serializable]
+    public class GrainReferenceNotBoundException : OrleansException
```

Comment:
Asked Sergey directly, and apparently we don't have that distinction... all exceptions that we generate typically inherit from OrleansException, so we are good here.